core/src/main/java/org/apache/iceberg/BaseRewriteFiles.java

```diff
@@ -57,4 +57,23 @@ public RewriteFiles rewriteFiles(Set<DataFile> filesToDelete, Set<DataFile> file
 
     return this;
   }
+
+  @Override
+  public RewriteFiles rewriteDeletes(Set<DeleteFile> deletesToDelete, Set<DeleteFile> deletesToAdd) {
+    Preconditions.checkArgument(deletesToDelete != null && !deletesToDelete.isEmpty(),
+        "Files to delete cannot be null or empty");
+    Preconditions.checkArgument(deletesToAdd != null && !deletesToAdd.isEmpty(),
+        "Files to add can not be null or empty");
+
+    for (DeleteFile toDelete : deletesToDelete) {
+      delete(toDelete);
```

Comment:
Nit:  for the replacing equality delete with position deletes,   we'd better to assert the delete files to be equality delete files ?  the delete file to add MUST be position delete files ?

Reply:
Make sense to me, but I'd like to keep logic here and assert in `ReplaceDeleteAction` since we may have an action to merge position delete files.